src/syscheckd/seechanges.c

```diff
@@ -569,7 +569,12 @@ int seechanges_estimate_compression(const float file_size) {
 }
 
 void seechanges_modify_estimation_percentage(const float compressed_size, const float uncompressed_size) {
-    float compression_rate = 1 - (compressed_size / uncompressed_size);
+    float compression_rate;
+
+    if (uncompressed_size <= 0) {
+        return;
+    }
+    compression_rate = 1 - (compressed_size / uncompressed_size);
```

Comment:
If `FileSize()` fails (`-1`) for the first parameter, `compression_rate` will be negative and the condition below will always match.